packages/file-asset-apis/src/s3/createS3Client.ts

```diff
@@ -30,56 +31,135 @@ export async function createS3Client(
+    intermediateConfig = moveCredentialsIntoObject(intermediateConfig);
+    intermediateConfig = mapMaxRetriesToMaxAttempts(intermediateConfig);
+    const finalConfig = removeExtendedConfigKeys(intermediateConfig);
+    return finalConfig;
+}
 
+/**
+ * Given the terafoundation S3 connector configuration, if sslEnabled is true,
+ * add CA certs and 'rejectUnauthorized: true' to httpOptions.
+ * @param {S3ClientConfig} startConfig terafoundation S3 connector configuration object
+ * @returns {Promise<S3ClientConfig>} AWS S3 client EXTENDED configuration object
+ */
+export async function addCertsIfSSLEnabled(startConfig: S3ClientConfig): Promise<S3ClientConfig> {
     // pull certLocation from env
     // https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
     // Instead of updating the client, we can just update the config before creating the client
-    if (config.sslEnabled) {
-        const certPath = config.certLocation ?? '/etc/ssl/certs/ca-certificates.crt';
+
+    let newOptions: HttpsAgentOptions = Object.assign(
+        startConfig.httpOptions || {}
+    );
+
+    if (startConfig.sslEnabled) {
+        const certPath = startConfig.certLocation ?? '/etc/ssl/certs/ca-certificates.crt';
         const pathFound = await fs.exists(certPath);
 
         if (!pathFound) {
             throw new Error(
-                `No cert path was found in config.certLocation: "${config.certLocation}" or in default "/etc/ssl/certs/ca-certificates.crt" location`
+                `No cert path was found in config.certLocation: "${startConfig.certLocation}" or in default "/etc/ssl/certs/ca-certificates.crt" location`
             );
         }
         // Assumes all certs needed are in a single bundle
         const certs = await fs.readFile(certPath);
 
-        httpOptions = {
-            ...httpOptions,
+        newOptions = {
+            ...newOptions,
             rejectUnauthorized: true,
             ca: [certs]
         };
     }
 
-    const { connectionTimeout, requestTimeout } = config.handlerOptions || {};
+    const endConfig = cloneDeep(startConfig);
+    endConfig.httpOptions = newOptions;
+    return endConfig;
+}
+
+/**
+ * Given the terafoundation S3 connector configuration, create a
+ * NodeHttpHandler and add it to the config
+ * @param {S3ClientConfig} startConfig terafoundation S3 connector configuration object
+ * @returns {Promise<S3ClientConfig>} AWS S3 client EXTENDED configuration object
+ */
+export function addRequestHandler(startConfig: S3ClientConfig): S3ClientConfig {
+    const { connectionTimeout, requestTimeout } = startConfig.handlerOptions || {};
     const requestHandlerOptions = {
         ...isNumber(connectionTimeout) && { connectionTimeout },
         ...isNumber(requestTimeout) && { requestTimeout },
-        ...!isEmpty(httpOptions) && new Agent(httpOptions)
+        ...!isEmpty(startConfig.httpOptions) && { httpsAgent: new Agent(startConfig.httpOptions) }
```

Comment:
This is the line with the bug fix. Should we be testing the value of `requestHandlerOptions`?